src/builder.rs

```diff
@@ -462,11 +462,20 @@ impl Builder {
                 .cloned()
                 .unwrap();
 
-            // use the first spend to create split spend(s) or create a dummy if empty.
-            let dummy_spend = spends.first().map_or_else(
-                || SpendInfo::dummy(asset, &mut rng),
-                |s| s.create_split_spend(),
-            );
+            // Extend the spends with dummy or split notes.
+            let dummy_spend = if asset.is_native().into() {
+                // For native asset, extends with dummy notes
+                SpendInfo::dummy(asset, &mut rng)
+            } else {
+                // For ZSA asset, extends with
+                // - dummy notes if first spend is empty
+                // - split notes otherwise.
+                spends.first().map_or_else(
+                    || SpendInfo::dummy(asset, &mut rng),
+                    |s| s.create_split_spend(),
+                )
+            };
+
```

Comment:
Unfortunately cloning `dummy_spend` will not work for the client because of having the same nullifier. We need a new dummy spend for each extend.

Reply:
consider wrapping `let dummy_spend = ..` into a function and trigger each time.

Reply:
Done